Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/Allocator/Jobs/DriverFeeUpdates/DriverFee.hs

```diff
@@ -68,34 +81,234 @@ sendPaymentReminderToDriver Job {id, jobInfo} = withLogTag ("JobId-" <> id.getId
+  let dutyDate = addUTCTime (secondsToNominalDiffTime transporterConfig.timeDiffFromUtc) driverFee.createdAt
+      registrationDateLocal = addUTCTime (secondsToNominalDiffTime transporterConfig.timeDiffFromUtc) registrationDate
+  if feeWithoutDiscount == 0
+    then do
+      updateStatus CLEARED now driverFee.id
+      return (0, 0, Nothing, Nothing)
+    else do
+      offers <- Payment.offerList merchantId (makeOfferReq feeWithoutDiscount driver plan dutyDate registrationDateLocal) -- handle UDFs
+      (finalOrderAmount, offerId, offerTitle) <-
+        if null offers.offerResp
+          then pure (baseAmount, Nothing, Nothing)
+          else do
+            let bestOffer = minimumBy (comparing (.finalOrderAmount)) offers.offerResp
+            pure (bestOffer.finalOrderAmount, Just bestOffer.offerId, bestOffer.offerDescription.title)
+      let (platformFee, cgst, sgst) = calculatePlatformFeeAttr finalOrderAmount plan -- this should be HighPrecMoney
+      updateFee driverFee.id Nothing 0 (round platformFee) cgst sgst now False -- add split logic before update
+      return (feeWithoutDiscount, finalOrderAmount, offerId, offerTitle)
+
+splitPlatformFee :: HighPrecMoney -> HighPrecMoney -> Plan -> DriverFee -> [DriverFee]
+splitPlatformFee feeWithoutDiscount_ totalFee plan DriverFee {..} =
+  let numEntities = totalFee / plan.maxAmount
+      remainingFee = totalFee `mod'` plan.maxAmount
+      entityList = replicate (floor numEntities) plan.maxAmount ++ [remainingFee | remainingFee > 0]
+   in map
+        ( \fee -> do
+            let (platformFee_, cgst, sgst) = calculatePlatformFeeAttr fee plan
+            DriverFee
+              { platformFee = PlatformFee {fee = round platformFee_, ..},
+                feeType = feeType,
+                feeWithoutDiscount = Just feeWithoutDiscount_, -- same for all splitted ones, not remaining fee
+                ..
+              }
+        )
+        -- govt_charges, num_rides, total_earnings are same for all these
+        entityList
+
+getFreqAndBaseAmountcase :: PlanBaseAmount -> (Text, HighPrecMoney)
+getFreqAndBaseAmountcase planBaseAmount = case planBaseAmount of
+  PERRIDE_BASE amount -> ("PER_RIDE" :: Text, amount)
+  DAILY_BASE amount -> ("DAILY" :: Text, amount)
+  WEEKLY_BASE amount -> ("WEEKLY" :: Text, amount)
+  MONTHLY_BASE amount -> ("MONTHLY" :: Text, amount)
+
+driverFeeSplitter :: (MonadFlow m) => Plan -> HighPrecMoney -> HighPrecMoney -> DriverFee -> UTCTime -> m ()
+driverFeeSplitter plan feeWithoutDiscount totalFee driverFee now = do
+  let splittedFees = splitPlatformFee feeWithoutDiscount totalFee plan driverFee
+  case plan.paymentMode of
+    MANUAL -> do
+      case splittedFees of
+        [] -> throwError (InternalError "No driver fee entity with non zero total fee")
+        (firstFee : restFees) -> do
+          updateStatus PAYMENT_OVERDUE now firstFee.id
+          updRestFees <- mapM (buildRestFees PAYMENT_OVERDUE RECURRING_INVOICE) restFees
```

Comment:
Make restFees as ONGOING so that these can be picked in next round of scheduler.